sled-agent/src/services.rs

```diff
@@ -148,14 +157,17 @@ impl ServiceManager {
         rack_id: Uuid,
     ) -> Result<Self, Error> {
         debug!(log, "Creating new ServiceManager");
+        let log = log.new(o!("component" => "ServiceManager"));
         let mgr = Self {
-            log: log.new(o!("component" => "ServiceManager")),
+            log: log.clone(),
             config,
             zones: Mutex::new(vec![]),
             vnic_allocator: VnicAllocator::new("Service", etherstub),
             underlay_vnic,
             underlay_address,
             rack_id,
+            ddmd_client: DdmAdminClient::new(log)?,
+            advertised_prefixes: Mutex::new(HashSet::new()),
```

Comment:
Just making sure I'm grokking this correctly, this is only going to be "GZ addresses from *our* sled", right?

(Just checking that "treat this as an addition-only datastructure" is reasonable, and it won't grow arbitrarily with a later addition of sled addition/removal)

Reply:
Correct, these are only the addresses for GZ services we start that we tell our mg-ddmd to advertise. We should withdraw any addresses we've advertised if we shut down the corresponding service (and also remove them from this set), but that would require https://github.com/oxidecomputer/maghemite/issues/9.